### PR TITLE
feat: improved handling for unevaluatedProperty errors

### PR DESCRIPTION
**Pull request by erunion**
## 🧰 Changes

This improves the handling of `unevaluatedProperty` errors that are now common with misplaced items in OpenAPI 3.1 schemas. Functionally here they're identical to `additionalProperties` errors, just with a different parameter.

## 🧬 QA & Testing

What this looked like before:

Screen Shot 2021-12-02 at 3 59 57 PM

And what it looks like now:

Screen Shot 2021-12-02 at 4 00 18 PM

### CODE REVIEW

File: src/validation-errors/__tests__/main.test.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import Ajv from 'ajv';
+import Ajv from 'ajv/dist/2020';
```
Comment (erunion) [inline src/validation-errors/__tests__/main.test.js:1]: `unevaluatedProperties` is new to the 2020 draft so for this new test to work I needed to change this import.

File: src/validation-errors/unevaluated-prop.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import BaseValidationError from './base';
+
+export default class UnevaluatedPropValidationError extends BaseValidationError {
```
Comment (ilias-t) [inline src/validation-errors/unevaluated-prop.js:3]: `UnevaluatedPropValidationError` is a bit of a mouthful; a comment explaining what it signifies somewhere would be nice 😬

Comment (ilias-t) [reply src/validation-errors/unevaluated-prop.js:3]: lol no docs here https://json-schema.org/understanding-json-schema/reference/object.html#unevaluated-properties

Comment (ilias-t) [reply src/validation-errors/unevaluated-prop.js:3]: Would `UnevaluatedPropertyError` work? or does it need to include validation for some reason?

Comment (erunion) [reply src/validation-errors/unevaluated-prop.js:3]: Unfortunately this is the same format for these error handling classes that's already present:

* RequiredValidationError
* AdditionalPropValidationError
* EnumValidationError
* DefaultValidationError
* UnevaluatedPropValidationError

**Review by Dashron — APPROVED** [review]
code looks good, color changes look good

**Pull request merged**